app/src/test/groovy/pl/futurecollars/invoicing/service/InvoiceServiceIntegrationTest.groovy

```diff
@@ -1,39 +1,35 @@
 package pl.futurecollars.invoicing.service
 
+import org.springframework.boot.test.context.SpringBootTest
 import pl.futurecollars.invoicing.db.Database
 import pl.futurecollars.invoicing.db.memory.InMemoryDatabase
-import pl.futurecollars.invoicing.model.Company
+import pl.futurecollars.invoicing.fixtures.InvoiceFixture
 import pl.futurecollars.invoicing.model.Invoice
-import pl.futurecollars.invoicing.model.InvoiceEntry
 import spock.lang.Specification
-import java.time.LocalDateTime
 import java.util.function.Predicate
 
+@SpringBootTest
 class InvoiceServiceIntegrationTest extends Specification {
 
-    Database database;
+    Database inMemoryDatabase;
     InvoiceService invoiceService;
-    Invoice invoice;
-    Company from = new Company(1L, "adress")
-    Company to = new Company(2L, "address")
-
+    Invoice invoice = InvoiceFixture.getInvoice()
     def setup() {
-        database = new InMemoryDatabase()
-        invoiceService = new InvoiceService(database)
-        invoice = new Invoice(LocalDateTime.now(), from, to, new ArrayList<InvoiceEntry>())
+        inMemoryDatabase = new InMemoryDatabase()

```

Comment:
why you got rid of generic type tests?

Reply:
I don't understand, I only moved the creation of Invoices to InvoiceFixture